### PR TITLE
Allow LazyRecord/LazyTable in JSON function

### PR DESCRIPTION
**Pull request by LucGenetier**

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by MikeStall — APPROVED** [review]

File: src/tests/Microsoft.PowerFx.Json.Tests/JSONTests.cs [hunk 1]
```diff
@@ -36,5 +37,52 @@ public void Json_IncludeBinaryData_NoSideEffects()
             Assert.False(result.IsSuccess);
             Assert.Equal("The JSON function cannot serialize binary data in non-behavioral expression.", result.Errors.First().Message);
         }
+
+        public class LazyRecordType : RecordType
```
Comment (lesaltzm) [inline src/tests/Microsoft.PowerFx.Json.Tests/JSONTests.cs:41]: Please add a test where there's a recursive type & value, validate that it errors after a certain depth

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Core/Public/Config/Features.cs [hunk 2]
```diff
@@ -119,6 +125,23 @@ internal static Features PowerFxV1AllowSetExpandedTypes
             CoalesceShortCircuit = true,
             PrimaryOutputPropertyCoercionDeprecated = true,
             AsTypeLegacyCheck = false,
+            JsonFunctionAcceptsLazyTypes = false
```
Comment (CarlosFigueiraMSFT) [inline src/libraries/Microsoft.PowerFx.Core/Public/Config/Features.cs:128]: >JsonFunctionAcceptsLazyTypes [](http://example.com/codeflow?start=12&length=28)

We can set it to true in PowerFxV1 - this won't break any existing scenarios (will only make current scenarios that don't work today work, so not a breaking change).

File: src/tests/Microsoft.PowerFx.Json.Tests/JSONTests.cs [hunk 3]
```diff
@@ -36,5 +37,122 @@ public void Json_IncludeBinaryData_NoSideEffects()
...
+            var record = RecordType.Empty();
+            record = record.Add("Property", new LazyRecordType());
+
+            var formulaParams = RecordType.Empty();
+            formulaParams = formulaParams.Add("Var", record);
+
+            var result = engine.Check("JSON(Var)", formulaParams);
+
+            Assert.True(result.IsSuccess);
+        }
```
Comment (CarlosFigueiraMSFT) [inline src/tests/Microsoft.PowerFx.Json.Tests/JSONTests.cs:78]: Can you add a test where we actually check the result of the JSON function, to make sure that we not only compile, but we properly serialize lazy records? Also a test with lazy tables would be good.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by CarlosFigueiraMSFT — APPROVED** [review]
:shipit:

File: src/libraries/Microsoft.PowerFx.Core/Public/Config/Features.cs [hunk 4]
```diff
@@ -87,6 +88,11 @@ public sealed class Features
         /// </summary>
         internal bool PrimaryOutputPropertyCoercionDeprecated { get; set; }
 
+        /// <summary>
+        /// This is specific for PVA team and it is a temporary feature.
+        /// </summary>
+        internal bool JsonFunctionAcceptsLazyTypes { get; set; }
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Public/Config/Features.cs:94]: We should not have this flag.  #2197

**Pull request merged**